lib/IRGen/IRGen.cpp

```diff
@@ -263,6 +263,8 @@ void swift::performLLVMOptimizations(const IRGenOptions &Opts,
                                           OptimizationLevel Level) {
       if (Level != OptimizationLevel::O0)
         MPM.addPass(createModuleToFunctionPassAdaptor(SwiftARCContractPass()));
+      if (Level == OptimizationLevel::O0)
+        MPM.addPass(AlwaysInlinerPass());
```

Comment:
Are we sure this (the instance added by this PR) is the only place the alwaysinline attribute is used? Because by adding an optimization pass at O0, we might be changing more than what we intended to change.

Reply:
Got it, thanks for the clarification!

Reply:
Also, at least as far as debugging is concerned, inlining is a 100% "reversible" operation, i.e., it should be transparent in the debugger.